db/reqlog.c

```diff
@@ -2004,7 +2004,7 @@ void log_long_running_sql_statements()
     Pthread_mutex_lock(&gbl_sql_lock);
     LISTC_FOR_EACH(&thedb->sql_threads, thd, lnk)
     {
-        if (thd->clnt != NULL) {
+        if (thd->clnt != NULL && thd->clnt->thd != NULL) {
```

Comment:
Code was never meant to be used from a different thread.  Each sql thread accesses its own fields, and knows not to do that after cleaning clnt structure.  With this, we are racing cleaning sql client with logging, since code frees first and then removes from list.